src/libs/ReportUtils.ts

```diff
@@ -1625,32 +1686,29 @@ function getTransactionDetails(transaction, createdDateFormat = CONST.DATE.FNS_F
  * - in case of expense report
  *    - the current user is the requestor and is not settled yet
  *    - or the user is an admin on the policy the expense report is tied to
- *
- * @param {Object} reportAction
- * @returns {Boolean}
  */
-function canEditMoneyRequest(reportAction) {
+function canEditMoneyRequest(reportAction: OnyxEntry<ReportAction>): boolean {
     const isDeleted = ReportActionsUtils.isDeletedAction(reportAction);
 
     if (isDeleted) {
         return false;
     }
 
     // If the report action is not IOU type, return true early
-    if (reportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
+    if (reportAction?.actionName !== CONST.REPORT.ACTIONS.TYPE.IOU) {
         return true;
     }
 
-    const moneyRequestReportID = lodashGet(reportAction, 'originalMessage.IOUReportID', 0);
+    const moneyRequestReportID = reportAction?.originalMessage?.IOUReportID ?? 0;
 
     if (!moneyRequestReportID) {
         return false;
     }
 
-    const moneyRequestReport = getReport(moneyRequestReportID);
-    const isReportSettled = isSettled(moneyRequestReport.reportID);
-    const isAdmin = isExpenseReport(moneyRequestReport) && lodashGet(getPolicy(moneyRequestReport.policyID), 'role', '') === CONST.POLICY.ROLE.ADMIN;
-    const isRequestor = currentUserAccountID === reportAction.actorAccountID;
+    const moneyRequestReport = getReport(String(moneyRequestReportID));
+    const isReportSettled = isSettled(moneyRequestReport?.reportID);
+    const isAdmin = ((isNotEmptyObject(moneyRequestReport) && isExpenseReport(moneyRequestReport) && getPolicy(moneyRequestReport?.policyID ?? '')?.role) ?? '') === CONST.POLICY.ROLE.ADMIN;
```

Comment:
```suggestion
    const isAdmin = isExpenseReport(moneyRequestReport) && (getPolicy(moneyRequestReport?.policyID ?? '').role ?? '') === CONST.POLICY.ROLE.ADMIN;
```
This should be better ig